### DISCUSSION TITLE
GO warning regarding ODF parameter Pipe999ReleaseCrossfadeLength

### DISCUSSION THREAD

**Discussion by pgwid-og** [GrandOrgue/grandorgue#1802] [Developers]
Let me start with a BIG THANK YOU for the recent developments around GrandOrgue!

Thanks to the very clever tool "OdfEdit" I started to migrate some of my HW-ODF to GrandOrgue :-) 

In some ODFs the parameter Pipe999ReleaseCrossfadeLength is used, which is described in the GO help. But when I load this ODF  in GO version 3.13.3-1 I get the warning, that this parameter is not supported. This is a bit annoying, since in big ODF this warning is thrown a few hundred times. Yes, I could switch off the "strict ODF" check, but still, I wonder why this happens. I even wrote a python script to eliminate the corresponding lines in the ODF to get rid of this parameter.

By the way:  With the "SP Casavant sample set" I  get a similar warning "Nicht verwendeter ODF Eintrag 'Rank031/Pipe042Attack001LoopCrossfadeLength' but a parameter Attack999LoopCrossfadeLenghth is not documented in GO-help, so this issue addresses the OdfEdit-Tool.

**Comment by eturpault** [comment]
I think this is because you are using OdfEdit v2.9 which implements new LoopCrossfadeLength and ReleaseCrossfadeLength attributes for the HW to GO conversion which will be supported from GrandOrgue v3.14.0, not yet an official version, as indicated in the OdfEdit release note : https://github.com/GrandOrgue/OdfEdit/releases/tag/v2.9

**Reply by pgwid-og** [reply]
Thank you, I guess you are right! I wasn't reading the warning carefully enough, it is indeed about "Pipe999Release99ReleaseCrossfadeLength" and not the old "Pipe999ReleaseCrossfadeLength".
So all is well with GO!

**Comment by eturpault** [comment]
Pipe999ReleaseCrossfadeLength is supported by GO since a long time. What will be new in GO 3.14.0 are the attributes Pipe999Attack999LoopCrossfadeLength, Pipe999Attack999ReleaseCrossfadeLength, Pipe999Release999ReleaseCrossfadeLength.

**Comment by larspalo** [comment]
The reason for the warnings is a bit more subtle than that, all because of the recent development of GO. But to resolve them, just re-generate any odf created from OdfEdit and all should be well! No original odf made for GO to begin with that I know of will be affected.

**Comment by sebion7125** [comment]
I am having trouble with the "Pipe999ReleaseCrossfadeLength" Parameter, which is used in the Sample Sets created by Piotr Grabowski. Even with strict ODF option disabled I am getting so many warnings, that a huge sample set literally gets impossible to load. Is it intentional that the old general Pipe999ReleaseCrossfadeLength Parameter is no longer supported?

**Reply by larspalo** [reply]
@sebion7125 Yes, it's very much intentional since the behaviour of GO has changed. There are however many ways to fix such issues, which all requires fixing the now misbehaving odf (if you want to use a recent version of GO). One possibility that you can test is using the GoOdf software with the following procedure:

- Create a backup copy of the original odf first
- Open the odf in GoOdf
- Use Tools->Import Legacy x-fades menu option
- Save the now modified .organ file by writing it out and test it

And for the record, "the old general Pipe999ReleaseCrossfadeLength Parameter" is still supported - but not in the same way as it was earlier. Now this _only_ apply to, and is valid to use for, a release that is also loaded from the Pipe999 attack sample (ie. the Pipe999LoadRelease=N cannot be used together with it).
